src/actions.sol

```diff
@@ -57,9 +71,45 @@ contract Actions {
     event ApproveNFT(address indexed registry, address indexed usr, uint256 tokenAmount);
     event ApproveERC20(address indexed erc20, address indexed usr, uint256 amount);
     event TransferERC20(address indexed erc20, address indexed dst, uint256 amount);
+    event Minted(address indexed registry, uint256 tokenId);
+
+    address public immutable shelf;
+    address public immutable pile;
+    address public immutable feed;
+    address public immutable self;
+
+    // address to deposit withdraws
+    address public immutable withdrawAddress;
+
+    // modifier only delegated call
+    modifier onlyDelegateCall() {
+        require(address(this) != self, "only-delegate-call");
+        _;
+    }
+
+    constructor(address root_, address withdrawAddress_) {
+        address shelf_ = BorrowerDeployerLike(RootLike(root_).borrowerDeployer()).shelf();
+        address pile_ = ShelfLike(shelf_).pile();
+        address feed_ = BorrowerDeployerLike(RootLike(root_).borrowerDeployer()).feed();
+        address self_ = address(this);
+        require(withdrawAddress_ != address(0), "withdraw-address-not-set");
+        require(shelf_ != address(0), "shelf-not-set");
+        require(pile_ != address(0), "pile-not-set");
+        require(feed_ != address(0), "feed-not-set");
+        withdrawAddress = withdrawAddress_;
+        shelf = shelf_;
+        self = self_;
+        pile = pile_;
+        feed = feed_;
+    }
+
+    function mintAsset(address registry) public onlyDelegateCall returns (uint256 tokenId) {
+        tokenId = NFTLike(registry).mintTo(address(this));
+        emit Minted(registry, tokenId);
```

Comment:
missing the return of the `tokenId` here

Reply:
The `tokenId` will be automatically returned because the name is defined in the function signature `returns (uint256 tokenId`). 

I also wrote a test for it `testMint`. It should work.

Reply:
haha okay my bad 😄